packages/ketcher-react/src/script/ui/views/modal/components/process/Miew/Miew.tsx

```diff
@@ -85,39 +98,19 @@ function createMiewOptions(userOpts) {
 const CHANGING_WARNING =
   'Stereocenters can be changed after the strong 3D rotation'
 
-class MiewDialog extends Component {
-  componentDidMount() {
-    const { struct, server, miewOpts } = this.props
-    const Miew = window.Miew
-
-    this.viewer = new Miew({
-      container: this.miewContainer
-    })
-
-    if (this.viewer.init()) this.viewer.run()
-
-    const factory = new FormatterFactory(server)
-    const service = factory.create('cml')
-
-    service
-      .getStructureFromStructAsync(struct)
-      .then((res) =>
-        this.viewer.load(res, { sourceType: 'immediate', fileType: 'cml' })
-      )
-      .then(() => this.viewer.setOptions(miewOpts))
-      .catch((ex) => console.error(ex.message))
-  }
+class MiewDialog extends Component<Props> {
```

Comment:
I think you could change it to a functional component as you almost fully refactored it